website/docs/maintenance/upgrading_postgres.md

```diff
@@ -3,28 +3,114 @@ title: Upgrading PostgreSQL to a new major version
 sidebar_label: Upgrading PostgreSQL
 ---
 
-1. Create a [backup](backup_restore.md)
-2. Stop all TeslaMate containers
+In these instructions it is assumed you are upgrading from postgresql 16 to postgresql 17. Please update
+references to 16 to your start version and 17 to your desired version.
+
+These instructions also assume you have the standard `docker-compose.yml` file. If this is not the case
+you might need to make adjustments. For example, older installed might call the container "db" not "database".
+
+:::note
+If you are using `docker-compose`, you are using Docker Compose v1, which has been deprecated. Docker Compose commands refer to Docker Compose v2. Consider upgrading your docker setup, see [Migrate to Compose V2](https://docs.docker.com/compose/migrate/)
+:::
+
+## Upgrade
+
+1. Create a backup:
+
+   ```bash
+   docker compose exec -T database pg_dump -U teslamate teslamate > teslamate_database_$(date +%F_%H-%M-%S).sql
+   ```
+
+2. _Check to make sure there are no obvious errors in this backup file._ There should not be any errors appearing
+   on screen. The file should be more than 0 bytes long. Do not proceed if there is any doubt.
+
+3. Stop all TeslaMate containers
 
    ```bash
    docker compose down
    ```
 
-3. Delete the database volume. **Be careful**, this will delete all your previously recorded data! Make sure that your backup can be restored before you start.
+4. Make a copy of your current `docker-compose.yml`
 
    ```bash
-   docker volume rm "$(basename "$PWD")_teslamate-db"
+   cp docker-compose.yml docker-compose-16.yml
```

Comment:
Why should users backup the compose file. We could say - in case of error please revert your changes to the compose file to get back to a working version (that should be fine to ask for).

Reply:
On one hand, I note that there is a comment that some environments may automatically delete any unexpected files, such as this backup. Oops.

On the other hand, not really convinced our users are good at following written instructions... Think spelling it out might be required.

Reply:
those would also delete the postgres backup, right?

as long as the data is still accessible everything is alright. if someone isn't able to follow a "revert changes made to docker-compose.yaml" file they still can open up a discussion / ask in forums for help.

Reply:
These will delete the updated `docker-compose.yml` for postgres. The postgres data itself should remain intact.

At least that is the theory. Possibly some platform might decide that since the data is no longer referenced it can be deleted. Hoping that doesn't actually happen...